lte/gateway/c/session_manager/test/test_set_session_manager_handler.cpp

```diff
@@ -672,6 +702,251 @@ TEST_F(SessionManagerHandlerTest, test_SetAmfSessionAmbr) {
+
+  insert_static_rule(1, "", "rule1");
+
+  CreateSessionResponse response;
+  auto credits = response.mutable_credits();
+  create_credit_update_response(IMSI1, SESSION_ID_1, 1, 4096, credits->Add());
+  SessionConfig cfg;
+  cfg.common_context       = request.common_context();
+  cfg.rat_specific_context = request.rat_specific_context();
+  cfg.rat_specific_context.mutable_m5gsm_session_context()->set_ssc_mode(
+      SSC_MODE_3);
+  grpc::ServerContext server_context;
+  set_session_manager->SetAmfSessionContext(
+      &server_context, &request,
+      [this](grpc::Status status, SmContextVoid Void) {});
+
+  // Run session creation in the EventBase loop
+  evb->loopOnce();
+  auto session_map = session_store->read_sessions({IMSI1});
+  session_enforcer->m5g_init_session_credit(
+      session_map, IMSI1, SESSION_ID_1, cfg);
+  SessionSearchCriteria criteria(IMSI1, IMSI_AND_SESSION_ID, SESSION_ID_1);
+  auto session_it = session_store->find_session(session_map, criteria);
+  auto& session_t = **session_it;
+  auto update     = SessionStore::get_default_session_update(session_map);
+  session_enforcer->update_session_with_policy(session_t, response, nullptr);
+  session_enforcer->m5g_update_session_context(
+      session_map, IMSI1, session_t, update);
+
+  assert_charging_credit(
+      session_map, IMSI1, SESSION_ID_1, ALLOWED_TOTAL, {{1, 4096}});
+
+  insert_static_rule(1, "1", "rule1");
+
+  RuleRecordTable table;
+  auto record_list = table.mutable_records();
+  create_rule_record(IMSI1, "rule1", 1024, 1024, record_list->Add());
+  // auto update = SessionStore::get_default_session_update(session_map);
+  auto uc = get_default_update_criteria();
+  session_map[IMSI1][0]->increment_rule_stats("rule1", &uc);
+  local_enforcer->aggregate_records(session_map, table, update);
+
+  session_store->create_sessions(IMSI1, std::move(session_map[IMSI1]));
+
+  UpdateSessionResponse update_response;
+  auto credit_updates_response = update_response.mutable_responses();
+  create_credit_update_response(
+      IMSI1, SESSION_ID_1, 1, 24, credit_updates_response->Add());
+
+  auto monitor_updates_response =
+      update_response.mutable_usage_monitor_responses();
+  create_monitor_update_response(
+      IMSI1, SESSION_ID_1, "1", MonitoringLevel::PCC_RULE_LEVEL, 2048,
+      monitor_updates_response->Add());
+
+  create_credit_update_response(
+      IMSI1, SESSION_ID_2, 1, 400, credit_updates_response->Add());
+  create_monitor_update_response(
+      IMSI1, SESSION_ID_2, "1", MonitoringLevel::PCC_RULE_LEVEL, 500,
+      monitor_updates_response->Add());
+
+  session_map = session_store->read_sessions(SessionRead{IMSI1});
+  local_enforcer->update_session_credits_and_rules(
+      session_map, update_response, update);
+
+  assert_charging_credit(
+      session_map, IMSI1, SESSION_ID_1, ALLOWED_TOTAL, {{1, 4120}});
+  assert_monitor_credit(
+      session_map, IMSI1, SESSION_ID_1, ALLOWED_TOTAL, {{"1", 2048}});
+}
+
```

Comment:
You modified  code on `aggregate_records`  function.  I may be wrong, but i do not see any test case covering that. I think that change should have triggered a new test  on `test_local_enforcer.cpp` to check that case  (see [here](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=lte%2Fgateway%2Fc%2Fsession_manager%2Ftest%2Ftest_local_enforcer.cpp&editor=JetBrains&version=v1.2.1&utm_product_name=CLion&utm_product_version=2021.1.2&start_row=400&start_col=31&end_row=400&end_col=31))

Reply:
Oh I see this was already mentioned in my coment 11 days ago
https://github.com/magma/magma/pull/10239/files#r747891970

Reply:
Already two test cases (test_single_record_5g and and test_update_session_credits_and_rules_5g) are included for ```aggregate_records``` and set the ```RATType```  value as a TGPP_NR. I have checked using gdb and hitting the code.